src/redisearch_rs/rqe_iterators/tests/integration/optional.rs

```diff
@@ -0,0 +1,1031 @@
+            .expect("read without error")
+            .expect("read some result, be it virtual or real");
+        let _ = it
+            .read()
+            .expect("read without error")
+            .expect("read some result, be it virtual or real");
+
+        // Revalidate should return VALIDATE_OK
+        let status = it.revalidate().expect("revalidate without error");
+        assert!(matches!(status, RQEValidateStatus::Ok));
+
+        // Verify child was revalidated
+        assert_eq!(data.revalidate_count(), 1);
+
+        // Should be able to continue reading
+        let _ = it
+            .read()
+            .expect("read without error after revalidate")
+            .expect("read some result after revalidate");
+    }
+
+    #[test]
+    fn test_revalidate_aborted() {
+        let (mut it, mut data) = setup_optional_iterator_with_mock_child_and_data();
+
+        // Child returns VALIDATE_ABORTED
+        data.set_revalidate_result(utils::MockRevalidateResult::Abort);
+
+        // Read a document first
+        let _ = it
+            .read()
+            .expect("read without error")
+            .expect("read some result, be it virtual or real");
+
+        // Optional iterator handles child abort gracefully by replacing with empty iterator
+        let status = it.revalidate().expect("revalidate without error");
+        assert!(matches!(status, RQEValidateStatus::Ok)); // Optional iterator continues even when child is aborted
+
+        // Should be able to continue reading (now all virtual hits)
+        let result = it
+            .read()
+            .expect("read without error after revalidate")
+            .expect("read some result after revalidate");
+        assert_eq!(result.weight, 0.);
+    }
+
+    #[test]
+    fn test_revalidate_moved() {
+        let (mut it, mut data) = setup_optional_iterator_with_mock_child_and_data();
+
+        // Child returns VALIDATE_MOVED
+        data.set_revalidate_result(utils::MockRevalidateResult::Move);
+
+        // Read to a real hit (document from child)
+        const DOC_ID: t_docId = 10;
+        match it
+            .skip_to(DOC_ID)
+            .expect("no error to be returned while skipping")
+        {
+            Some(SkipToOutcome::Found(result)) => {
+                assert_eq!(result.doc_id, DOC_ID);
+            }
+            outcome @ (None | Some(SkipToOutcome::NotFound(_))) => {
+                panic!("unexpected outcome: {outcome:?}");
+            }
+        }
+        assert_eq!(it.last_doc_id(), DOC_ID);
+
+        // Revalidate should handle child movement
+        let status = it.revalidate().expect("revalidate without error");
+        // Should either be OK (if virtual result) or MOVED (if real result was affected)
+        assert!(matches!(
+            status,
+            RQEValidateStatus::Ok | RQEValidateStatus::Moved { .. }
+        ));
+
+        // Should be able to continue reading after revalidation
+        let _ = it
```

Comment:
Consider testing, after the re-validation, that first the current ID is 11, and then the read returns 12